### PR TITLE
Ports and Tweaks CM Push-ups

### PR DESCRIPTION
**Pull request by Geevies**
* Ported push-ups from CM. You can initiate them from the IC tab by hitting Do Pushup.

Ported (and tweaked) from https://github.com/cmss13-devs/cmss13/pull/876

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

File: code/modules/mob/living/carbon/human/species/species.dm [hunk 1]
```diff
@@ -830,3 +830,10 @@
 
 /datum/species/proc/can_double_fireman_carry()
 	return FALSE
+
+/datum/species/proc/has_stamina_for_pushup(var/mob/living/carbon/human/human)
```
Comment (JohnWildkins) [inline code/modules/mob/living/carbon/human/species/species.dm:834]: I think i'd rather this be a generic has_stamina proc for X amount of stamina drain, but w.e

**Review by JohnWildkins — APPROVED** [review]

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	do_pushups()
+
+/mob/living/carbon/human/proc/do_pushups()
+	if(!can_do_pushup())
+		return
+
+	set_face_dir(WEST)
+	visible_message(SPAN_NOTICE("[src] gets down and prepares to do some pushups."), SPAN_NOTICE("You get down for some pushups."), SPAN_NOTICE("You hear rustling."))
+
+	switch(alert(src, "Proper pushups or on your knees?", "What kinda pushups, mate.", "Proper ones", "On my knees"))
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:19]:
```suggestion
	switch(alert(src, "Regular pushups or on your knees?", "Pushups", "Regular", "On Knees"))
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		if("Proper ones")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:20]:
```suggestion
		if("Proper Ones")
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+			visible_message(SPAN_NOTICE("[src] shifts [get_pronoun("his")] weight onto [get_pronoun("his")] hands and feet."), SPAN_NOTICE("You move your weight onto your hands and feet."), SPAN_NOTICE("You hear rustling."))
+			execute_pushups(on_knees = FALSE)
+		if("On my knees")
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:23]:
```suggestion
		if("On Knees")
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+			visible_message(SPAN_NOTICE("[src] shifts [get_pronoun("his")] weight onto [get_pronoun("his")] knees. What a wimp."), SPAN_NOTICE("You move your weight onto your knees. WEAK!"), SPAN_NOTICE("You hear rustling."))
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:24]:
```suggestion
			visible_message(SPAN_NOTICE("[src] shifts [get_pronoun("his")] weight onto [get_pronoun("his")] knees."), SPAN_NOTICE("You move your weight onto your knees."), SPAN_NOTICE("You hear rustling."))
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+		species.drain_stamina(src, calculate_stamina_loss_per_pushup(on_knees))
+	to_chat(src, SPAN_WARNING("You slump down to the floor, too tired to keep going."))
+	stop_pushups()
+
+/mob/living/carbon/human/proc/can_do_pushup()
+	if(incapacitated(INCAPACITATION_RESTRAINED|INCAPACITATION_STUNNED|INCAPACITATION_KNOCKOUT))
+		return FALSE
+
+	if(!resting)
+		to_chat(src, SPAN_WARNING("How do you think you'll be able to do a pushup standing up? Get down to the floor!"))
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:66]:
```suggestion
		to_chat(src, SPAN_WARNING("You need to lie on the floor to do a pushup."))
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+		return FALSE
+
+	if(buckled_to)
+		to_chat(src, SPAN_WARNING("How do you think you'll be able to do a pushup while buckled to something? Get down to the floor!"))
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:70]:
```suggestion
		to_chat(src, SPAN_WARNING("You need to lie on the floor to do a pushup."))
```

File: code/modules/mob/living/carbon/human/exercise.dm [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+		return FALSE
+
+	var/list/extremities = list(BP_L_HAND, BP_R_HAND, BP_L_FOOT, BP_R_FOOT, BP_L_ARM, BP_R_ARM, BP_L_LEG, BP_R_LEG)
+	for(var/zone in extremities)
+		if(!get_organ(zone, TRUE))
+			to_chat(src, SPAN_WARNING("How do you think you'll be able to do a pushup without two hands and feet to stand on? See a doctor!"))
```
Comment (SleepyGemmy) [inline code/modules/mob/living/carbon/human/exercise.dm:76]:
```suggestion
			to_chat(src, SPAN_WARNING("You can't do a pushup with missing limbs."))
```

**Review by SleepyGemmy — CHANGES_REQUESTED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**